src/core/lib/security/credentials/oauth2/oauth2_credentials.cc

```diff
@@ -392,12 +392,11 @@ class grpc_compute_engine_token_fetcher_credentials
         const_cast<char*>(GRPC_COMPUTE_ENGINE_METADATA_TOKEN_PATH);
     request.http.hdr_count = 1;
     request.http.hdrs = &header;
-    /* TODO(ctiller): Carry the resource_quota in ctx and share it with the host
+    /* TODO(ctiller): Carry the memory quota in ctx and share it with the host
```

Comment:
(This comment isn't blocking this PR; it's just a question about strategy for subsequent changes.)

One possible way of doing this would be to have the channel pass its RQ to the [`grpc_call_credentials::get_request_metadata()` method](https://github.com/grpc/grpc/blob/790ce5c97ff3a3922d50a1fd3a4bab91e2bcd435/src/core/lib/security/credentials/credentials.h#L186), so that each attempt to get call creds can be charged to the channel that requests it.

However, the oauth case demonstrates that this might not be ideal, since the same call creds object can be used by multiple channels, and each channel may have its own RQ.  That would mean that the first channel that happens to trigger fetching the oauth token would be charged for the memory usage, and calls from another channel that start while the oauth fetch is pending will just block waiting for the fetch to complete without being charged for it.

Another option might be to add the ability to pass an RQ to use directly into the call credential when it is instantiated, so that the application can directly control which RQ is used by the call cred, regardless of what channels the call cred is used in.